app/api/server/v1/voip/server-config.ts

```diff
@@ -18,6 +20,37 @@ API.v1.addRoute('voipServerConfig.management', { authRequired: true }, {
 
 		return API.v1.success({ ...config });
 	},
+	// NOTE: you can use this POST endpoint for both create and update operation
+	post() {
+		check(this.bodyParams, Match.ObjectIncluding({
+			host: String,
+			port: Number,
+			serverName: String,
+			username: String,
+			password: String,
+		}));
+
+		if (!hasPermission(this.userId, 'manage-voip-contact-center-settings')) {
+			return API.v1.unauthorized('error-no-permission-manage-voip-contact-center-settings'); // TODO: create translations
+		}
+
+		const { host, port, serverName, username, password } = this.bodyParams;
+
+		Promise.await(Voip.deleteServerConfigDataIfAvailable(ServerType.MANAGEMENT));
```

Comment:
I have some questions about how we intend to handle update/delete. 
1. This server information is going to be a part of our reports. If we delete (or update) the record completely, we will not have a way to trace back the server information from the report if someone deletes/modifies the record.
2. The evidence trail is important in case of auditing. So data actually does not get deleted for predefined duration. 
It is usually 3 years.   
So we need to keep in mind that while preparing the report, we actually duplicate the entire server record instead of just the server name. Can we add a before |deleteServerConfigDataIfAvailable| to document this fact?

Reply:
Hi, @amolghode1981 Thanks for the feedback. Based on the use-case u shared, I think we'll need to update the DB schema as well to maintain the audit trail. However, before modifying the schema, I wanted to make sure I understand the overall requirements first, so can you please answer the following questions

1. Do we need to keep a track of users who created/edited/deleted a management config record?
2. Do we also need to keep a track of `call-server` type?
3. Will in the reports, there will ever be a query like "Show me server records active during particular time-frame?"

Reply:
More info about point 3: This question is in the context of whether we need to store data about the duration of the record when it was active

Reply:
> Do we need to keep a track of users who created/edited/deleted a management config record?
No. We do not need to keep track of users who did the operations.
> Do we also need to keep a track of call-server type records?
We need to keep track of both. Management type record is used only to pull the data but the data dictates the behaviour of call server. E.g We will store that agent x served queue y on call server z. If the call server is changed to w which does not have a queue y, this record will not make sense if we do not maintain the call server reference. I think same holds true for management interface.
> Will in the reports, there will ever be a query like "Show me server records active during particular time-frame?", More info about point 3: This question is in the context of whether we need to store data about the duration of the record when it was active. 
I dont think that would be required. While inserting the data in the report datacollection, we will have to put the timestamp because most of the reports are going to be trend based reports (e.g Call served per agent per month or abandoned calls per month)

I am not sure if we need to modify the schema at this point of time. in CE, we do not have any reports or schemas related to it. Neither for CEI suggest that it would be good to just add a comment around deleteServerConfigDataIfAvailable to consider this fact.